compat/os/mingw.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_OS_WINDOWS_MINGW_H
```

Comment:
The code changes in this PR appear to treat MinGW as a stand-alone OS rather than just a Windows "flavor". If that is the design we are going forward with, then we should name this guard consistent with guards in all other similar OS portability headers: `SQUID_OS_MINGW_H`.